crates/core/src/db/messages/commit.rs

```diff
@@ -1,81 +1,117 @@
+use anyhow::{bail, Context as _};
+use spacetimedb_sats::buffer::{BufReader, BufWriter};
+use std::{fmt, sync::Arc};
+
 use super::transaction::Transaction;
 use crate::hash::Hash;
-use std::sync::Arc;
 
-// aka "Block" from blockchain, aka RecordBatch, aka TxBatch
-#[derive(Debug)]
+#[cfg(test)]
+use proptest::prelude::*;
+#[cfg(test)]
+use proptest_derive::Arbitrary;
+
+/// A commit is one record in the write-ahead log.
```

Comment:
A drive by thought: similarly to how we have a bunch of code that says `set_id` but commented `"this is the table id"`, I would love to settle on saying "write ahead log" everywhere, including data type names etc. Or some other name. But let's pick one!

Reply:
Yeah, I'd love to settle on WAL / write ahead log, maybe even collapsing `CommitLog` and `MessageLog` into a single type. Possibly not for this PR, though?

Reply:
Yeah I'd love to have us try to take things that way. I agree on the naming, I think the commit log and message log names are implementation details and that "write ahead log" is the name that reflects the semantics the datastore needs to care about. But yeah not for this PR.

Reply:
Just adding my two cents here. I agree on coalescing on the name. I avoided WAL because historically that's referred to a log that is eventually compressed/deleted, but I think on balance WAL is probably the best name. We can include a comment about the fact that the WAL should never be deleted.

We should also note that it will include information which is not normally in a WAL, including reducer call event info and that it forms a Merkle DAG between commits.

Reply:
The term `MessageLog` was borrowed from Kafka.

Reply:
Noted to include all of these in the upcoming formalization / design changes doc.